django/db/backends/utils.py

```diff
@@ -135,8 +135,10 @@ def typecast_timestamp(s):  # does NOT store time zone information
     # "2005-07-29 09:56:00-05"
     if not s:
         return None
-    if ' ' not in s:
+    if ':' not in s:
         return typecast_date(s)
+    if '-' not in s:
+        return typecast_time(s)
```

Comment:
I haven't investigated if it's a possible input as we don't deal with time zone aware times but it looks like this could fail casting `'09:56:00-05'` to a `time`. Maybe we should use `if ' ' not in s` just to be sure here?